src/main/java/carpet/mixins/StructureBlockBlockEntityMixin.java

```diff
@@ -0,0 +1,77 @@
+package carpet.mixins;
+
+import carpet.CarpetSettings;
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.minecraft.block.Block;
+import net.minecraft.block.entity.StructureBlockBlockEntity;
+import net.minecraft.structure.Structure;
+import net.minecraft.structure.StructurePlacementData;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.ServerWorldAccess;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.*;
+
+import java.util.Random;
+
+@Mixin(StructureBlockBlockEntity.class)
+public abstract class StructureBlockBlockEntityMixin
+{
+    @Redirect(method = "place", at = @At(
+            value = "INVOKE",
+            target = "Lnet/minecraft/structure/Structure;place(Lnet/minecraft/world/ServerWorldAccess;Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/structure/StructurePlacementData;Ljava/util/Random;)V"
+    ))
+    private void onStructurePlacen(Structure structure, ServerWorldAccess serverWorldAccess, BlockPos pos, StructurePlacementData placementData, Random random)
+    {
+        if(!CarpetSettings.fillUpdates)
+            CarpetSettings.impendingFillSkipUpdates = true;
+        try
+        {
+            structure.place(serverWorldAccess, pos, placementData, random);
+        }
+        finally
+        {
+            CarpetSettings.impendingFillSkipUpdates = false;
+        }
+    }
+
+
+    @ModifyConstant(
+            method = "fromTag",
+            constant = @Constant(intValue = 48)
+    )
+    private int positiveLimit(int original) {
```

Comment:
If there are other mods already that modify that constant - its a guaranteed mod incompatibility. Its fine if you are not aware of any of these, so its just a note, not a request to change.

Reply:
Well it is unlikely for an other mod to change that exact value as it is only used for structure block reading. I can't see any better solution for that problem so I will leave it as it is for now.